cilium/error.go

```diff
@@ -0,0 +1,202 @@
+
+	// ErrNilClient represents the error which is
+	// generated when kubernetes client is nil
+	ErrNilClient = errors.New(ErrNilClientCode, errors.Alert, []string{"kubernetes client not initialized"}, []string{"Kubernetes client is nil"}, []string{"kubernetes client not initialized"}, []string{"Reconnect the adaptor to Meshery server"})
+
+	// ErrParseOAMComponent represents the error which is
+	// generated during the OAM component parsing
+	ErrParseOAMComponent = errors.New(ErrParseOAMComponentCode, errors.Alert, []string{"error parsing the component"}, []string{"Error occured while prasing application component in the OAM request made"}, []string{"Invalid OAM component passed in OAM request"}, []string{"Check if your request has vaild OAM components"})
+
+	// ErrParseOAMConfig represents the error which is
+	// generated during the OAM configuration parsing
+	ErrParseOAMConfig = errors.New(ErrParseOAMConfigCode, errors.Alert, []string{"error parsing the configuration"}, []string{"Error occured while prasing component config in the OAM request made"}, []string{"Invalid OAM config passed in OAM request"}, []string{"Check if your request has vaild OAM config"})
+
+	// ErrGetLatestReleaseCode represents the error which is
+	// generated when the latest stable version could not
+	// be fetched during runtime component registeration
+	ErrGetLatestReleaseCode = "1020"
+
+	//ErrLoadNamespaceCode occur during the process of applying namespace
+	ErrLoadNamespaceCode = "1024"
+)
+
+// ErrInstallCilium is the error for install mesh
+func ErrInstallCilium(err error) error {
+	return errors.New(ErrInstallCiliumCode, errors.Alert, []string{"Error with Cilium operation"}, []string{"Error occured while installing Cilium mesh through Ciliumctl", err.Error()}, []string{}, []string{})
+}
+
+// ErrTarXZF is the error for unzipping the file
+func ErrTarXZF(err error) error {
+	return errors.New(ErrTarXZFCode, errors.Alert, []string{"Error while extracting file"}, []string{err.Error()}, []string{"The gzip might be corrupt"}, []string{"Retry the operation"})
+}
+
+// ErrMeshConfig is the error for mesh config
+func ErrMeshConfig(err error) error {
+	return errors.New(ErrMeshConfigCode, errors.Alert, []string{"Error configuration mesh"}, []string{err.Error(), "Error getting MeshSpecKey config from in-memory configuration"}, []string{}, []string{"Reconnect the adaptor to the meshkit server"})
+}
+
+// ErrRunCiliumCtlCmd is the error for mesh port forward
+func ErrRunCiliumCtlCmd(err error, des string) error {
+	return errors.New(ErrRunCiliumCtlCmdCode, errors.Alert, []string{"Error running istioctl command"}, []string{err.Error()}, []string{"Corrupted istioctl binary", "Command might be invalid"}, []string{})
```

Comment:
Remove "istio". Also dont replace with ciliumctl, there is no such thing as cilium ctl. There CLI is just "cilium"

Reply:
Updated it 👍